src/cljs/main/broadfcui/page/workspace/details.cljs

```diff
@@ -108,10 +108,13 @@
                   (swap! state assoc :bucket-status-code status-code :bucket-access? success?))})
      (endpoints/call-ajax-orch
       {:endpoint (endpoints/get-workspace (:workspace-id props))
-       :on-done (fn [{:keys [success? get-parsed-response status-text]}]
+       :on-done (fn [{:keys [success? get-parsed-response xhr]}]
                   (if success?
                     (swap! state assoc :workspace (process-workspace (get-parsed-response)))
-                    (swap! state assoc :workspace-error status-text)))}))
+                    (swap! state assoc :workspace-error (as-> xhr x
+                                                            (aget x "responseText")
+                                                            (js-invoke js/JSON "parse" x)
+                                                            (aget x "message")))))}))
```

Comment:
I'm not crazy about `get-parsed-response` because it's tricky to see the function signature to understand how it works. In either case, there is one more step that needs to happen, which is handling the potential parse error if/when the error response is garbage. I'm also okay with simply ignoring that case since it is likely to be rare.